dandi/dandiapi.py

```diff
@@ -1023,7 +1023,7 @@ def wait_until_valid(self, max_time: float = 120) -> None:
         start = time()
         while time() - start < max_time:
             v = self.get_version(self.version_id)
-            if v.status is VersionStatus.VALID:
+            if v.status is VersionStatus.VALID and not v.asset_validation_errors:
```

Comment:
@danlamanna and @AlmightyYakob -- would you be so kind to digest for me here why having "Valid" indicator is not sufficient any longer and we also check that there is no errors?  related: https://github.com/dandi/dandisets/issues/356#issuecomment-1687321702

Reply:
"Valid" used to be returned only if both the version metadata and all of the assets were valid. The behavior has since been changed to only represent the version metadata, and so the extra check is required.